### ISSUE TITLE
[Bug]: Exeter City Council date time bug

### ISSUE THREAD

**Issue by schklom** [mampfes/hacs_waste_collection_schedule#1642]
### I Have A Problem With:

A specific source

### What's Your Problem

The integration does not work. It seems that `re.compile(REGEX_ORDINALS).sub("", d.text)` returns an empty string for some reason.

https://github.com/mampfes/hacs_waste_collection_schedule/blob/d57d1549f16371674a4c76c4224ced7f34870c00/custom_components/waste_collection_schedule/waste_collection_schedule/source/exeter_gov_uk.py#L47-L49

### Source (if relevant)

exeter_gov_uk

### Logs

```Shell
Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136
Integration: waste_collection_schedule (documentation)
First occurred: 14:27:55 (2 occurrences)
Last logged: 14:37:49
fetch failed for source Exeter City Council: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/exeter_gov_uk.py", line 47, in fetch date=datetime.strptime( ^^^^^^^^^^^^^^^^^^ File "/usr/lib/python3.11/_strptime.py", line 568, in _strptime_datetime tt, fraction, gmtoff_fraction = _strptime(data_string, format) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/lib/python3.11/_strptime.py", line 349, in _strptime raise ValueError("time data %r does not match format %r" % ValueError: time data '' does not match format '%A, %d %B %Y'
```

### Relevant Configuration

```YAML
waste_collection_schedule:
  sources:
    - name: exeter_gov_uk
      args:
        uprn: "10023120282"

sensor:
  - platform: waste_collection_schedule
    name: My_garbage
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [ ] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [ ] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by ReneNulschDE** [comment]
Reason: No date is delivered for "garden waste collection" with your upnr
```
b:  Refuse collection
d:  Monday, 29th January 2024
b:  Recycling collection
d:  Monday, 5th February 2024
b:  Garden waste collection
d:  
```
I will create a fix and PR.

**Comment by 5ila5** [comment]
Fixed in #1710

This will be part of release 1.46.0 or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
